kotlin-oauth-authenticator-archetype/src/main/resources/archetype-resources/src/main/kotlin/authentication/__pluginName__AuthenticatorRequestHandler.kt

```diff
@@ -0,0 +1,84 @@
+package ${package}.authentication
+
+import ${package}.config.${pluginName}AuthenticatorPluginConfig
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import se.curity.identityserver.sdk.attribute.Attribute
+import se.curity.identityserver.sdk.authentication.AuthenticationResult
+import se.curity.identityserver.sdk.authentication.AuthenticatorRequestHandler
+import se.curity.identityserver.sdk.errors.ErrorCode
+import se.curity.identityserver.sdk.http.RedirectStatusCode
+import se.curity.identityserver.sdk.service.ExceptionFactory
+import se.curity.identityserver.sdk.service.authentication.AuthenticatorInformationProvider
+import se.curity.identityserver.sdk.web.Request
+import se.curity.identityserver.sdk.web.Response
+
+import java.net.MalformedURLException
+import java.net.URL
+import java.util.Optional
+import java.util.UUID
+
+import ${package}.descriptor.${pluginName}AuthenticatorPluginDescriptor.Companion.CALLBACK
+
+class ${pluginName}AuthenticatorRequestHandler(private val _config: ${pluginName}AuthenticatorPluginConfig): AuthenticatorRequestHandler<Request>
+{
+    private val _authenticatorInformationProvider: AuthenticatorInformationProvider = _config.getAuthenticatorInformationProvider()
+    private val _exceptionFactory: ExceptionFactory = _config.getExceptionFactory()
+
+    override fun get(request: Request, response: Response): Optional<AuthenticationResult>
+    {
+        _logger.debug("GET request received for authentication")
+
+        val redirectUri = createRedirectUri()
+        val state = UUID.randomUUID().toString()
+        val scopes = setOf("scope")
+
+        _config.getSessionManager().put(Attribute.of("state", state))
+
+        val queryStringArguments = linkedMapOf<String, Collection<String>>(
+                Pair("client_id", setOf(_config.getClientId())),
```

Comment:
This is supposed to be prettier by using `"client_id" to setOf(_config.getClientId())`.

Reply:
Nice, I didn't know this one, thanks 👍